glusterd2/commands/peers/addpeer.go

```diff
@@ -80,7 +80,11 @@ func addPeerHandler(w http.ResponseWriter, r *http.Request) {
 	// Get the new peer information to reply back with
 	newpeer, err := peer.GetPeer(rsp.PeerID)
 	if err != nil {
-		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "new peer was added, but could not find peer in store. Try again later.")
+		if err == errors.ErrPeerNotFound {
```

Comment:
This isn't valid change here. Peer not found is an incorrect response to return to client for an add peer operation.